src/utils/api-request.ts

```diff
@@ -459,6 +470,11 @@ class AsyncHttpCall {
     try {
       this.config = new HttpRequestConfigImpl(config);
       this.options = this.config.buildRequestOptions();
+      if (!this.options.headers) {
```

Comment:
This doesn't look right. It is possible for this to be undefined. May be initialize to empty object?

Reply:
I see. The method `buildEntity()` also has code that modifies headers in-place.  I think we can just do this:

```
this.entity = this.config.buildEntity(this.options.headers!);
```

If `buildRequestOptions` did not set headers, this will trigger a runtime error in `buildEntity()` when it attempts to modify headers. That sounds like an ok position for us to be in. WDYT?